### PR TITLE
Fix for vspc-143

### PR DESCRIPTION
**Pull request by ashmichheda**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
On the public space page should be a full width option
## Anything else the reviewer needs to know?
Added checkbox option to resize the image width

### CODE REVIEW

**Comment by jdamerow** [comment]
top part of PR is not completely filled out.

**Comment by jdamerow** [comment]
The menu icon and image need to be closer to the top. The module page is fine but for spaces, the menu icon should be in the left top corner.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 1]
```diff
@@ -73,6 +73,24 @@
...
+                // resize the image to max width
+				if(checkBox.checked) {
+				    $("#bgImage").css("width", maxImgWidth);
+				}
+				// set the image width to default
+				else{
+                    $("#bgImage").css("width", spaceWidth);
+                    $('#bgImage').width('100%')
+                }
+			});
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:92]: indentation

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- the link positions need to be recalculated to stay in the same spots as before.
- let's make the button for fullscreen an icon maybe in the right upper corner (talk to Adhvik, he had something like that in his initial mockup)

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 2]
```diff
@@ -73,6 +73,28 @@
     <div layout:fragment="content">
         <script th:inline="javascript">
 		//# sourceURL=click.js
+		
+            let flag = 1;
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:77]: bad variable name

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 3]
```diff
@@ -73,6 +73,28 @@
...
+            function toggleImgSize(){
+                let maxImgWidth = '1200px';
+                // resize the image to max width
+			    if (flag == 1) {
+                    flag = -flag;
+                    document.getElementIById('toggleBtn').className = "icon-collapse";
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:83]: let's use jquery like the rest of the script

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- resize button doesn't seem to work
- button should ahve the same style as all the other links/buttons on the page
- remove "Resize image" text and show that as a tooltip
- it looks like all links on the image are moved down to accommodate the button

**Comment by ashmichheda** [comment]
> * resize button doesn't seem to work
> * button should ahve the same style as all the other links/buttons on the page
> * remove "Resize image" text and show that as a tooltip
> * it looks like all links on the image are moved down to accommodate the button

Fixed the above review comments

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- icon still doesn't look like the other icons (ask Adhvik for help)
- it stills moves everything down

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 4]
```diff
@@ -73,6 +102,29 @@
...
 		//# sourceURL=click.js
+		
+            let isResizeImgBtnClicked = 1;
+            function toggleImgSize(){
+                let maxImgWidth = '1200px';
+                let spaceWidth = $("#bgImage").css("width");
+                // resize the image to max width
+			    if (isResizeImgBtnClicked == 1) {
+			    	isResizeImgBtnClicked = -isResizeImgBtnClicked;
+			    	$("#toggleImgClass").attr('class', "icon-collapse secondary");
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:113]: indentation

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 5]
```diff
@@ -73,6 +102,29 @@
     <div layout:fragment="content">
         <script th:inline="javascript">
 		//# sourceURL=click.js
+		
+            let isResizeImgBtnClicked = 1;
+            function toggleImgSize(){
+                let maxImgWidth = '1200px';
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:108]: it should take the maximum possible width given window size (so not bigger than window size and as big as possible as long as it's not artificially blowing up the image).

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Let's make the circle with the full screen icon a little smaller so there is less white space around the icon.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 4 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:174]: ?

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
The image isn't shown correctly after making it fullscreen. Also, the button to make it smaller doesn't work. And it needs to remember the fullscreen setting so that the user does not need to set it on every page.
image

**Comment by ashmichheda** [comment]
> The image isn't shown correctly after making it fullscreen. Also, the button to make it smaller doesn't work. And it needs to remember the fullscreen setting so that the user does not need to set it on every page.
> image


- Since you had suggested to use the window's width, I am fetching the window's default width and setting it. Do you want me to assign some other default maximum width size?
- I didn't understand the other point that it needs to remember the full-screen setting? Once the page reloads it is set back to default.

**Comment by jdamerow** [comment]
- Window width is fine, but right now the image is cut off on the left. 
- When I make it full screen and go to another space, it should stay full screen until I change it back.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 4 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:178]: there are still indentation issues here

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 6]
```diff
@@ -134,6 +164,29 @@
...
+                    $("#bgImage").css("width", maxImgWidth);
+                    $("#space a").remove();
+                }
+                else{
+                	$("#toggleImgClass").attr('class', "icon-expand secondary");
+                    $("#bgImage").css("width", spaceWidth);
+                    $('#bgImage').width('100%')
+                    $("#space a").remove();
+                }
+			    drawLinks();
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:188]: and here

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 7]
```diff
@@ -134,6 +164,29 @@
     <div layout:fragment="content">
         <script th:inline="javascript">
 		//# sourceURL=click.js
+			
+            function toggleImgSize(isClicked){
+                let maxImgWidth = $(window).width();
+                let spaceWidth = $("#bgImage").css("width");
+                isResizeImgBtnClicked = getCookie('isResizeImgBtnClicked');
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:171]: variable should be declared

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 8]
```diff
@@ -172,14 +225,39 @@
...
+		    for(var i=0;i < ca.length;i++) {
+		        var c = ca[i];
+		        while (c.charAt(0)==' ') c = c.substring(1,c.length);
+		        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
+		    }
+		    return null;
+		}
+		
 		$(window).on("load", function() {
+			isResizeImgBtnClicked = getCookie('isResizeImgBtnClicked');
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:250]: variable should be declared

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
image is still cut-off on the left side
image

**Comment by pradeep-017** [comment]
Fixed Image cutoff issue and persisting the resize option of the image across spaces has been working too.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: vspace/pom.xml [hunk 9]
```diff
@@ -10,7 +10,7 @@
 
     <properties>
         <org.slf4j-version>1.7.5</org.slf4j-version>
-        <spring.version>5.0.19.RELEASE</spring.version>
+        <spring.version>5.0.7.RELEASE</spring.version>
```
Comment (jdamerow) [inline vspace/pom.xml:13]: why are you downgrading this?

File: vspace/pom.xml [hunk 10]
```diff
@@ -23,10 +23,10 @@
...
 
         <db.driver>com.mysql.jdbc.Driver</db.driver>
-        <db.url>jdbc:mysql://localhost:3306/vspace?serverTimezone=UTC</db.url>
-        <db.user>vspace</db.user>
-        <db.password>vspace</db.password>
-        <uploaded.files.path></uploaded.files.path>
+        <db.url>jdbc:mysql://localhost:3307/vspace?serverTimezone=UTC</db.url>
+        <db.user>root</db.user>
+        <db.password>root</db.password>
+        <uploaded.files.path>/Users/pradeep/DeveloperAtDiging/storage-docker</uploaded.files.path>
```
Comment (jdamerow) [inline vspace/pom.xml:29]: don't change this values. these should all be set via settins.xml not. changed in the pom.

File: vspace/pom.xml [hunk 11]
```diff
@@ -186,6 +186,17 @@
...
         </dependency>
+        <!-- Added by Pradeep  -->
+        <dependency>
+		  <groupId>javax.annotation</groupId>
+		  <artifactId>javax.annotation-api</artifactId>
+		  <version>1.3.2</version>
+		</dependency>
+		<dependency>
+		    <groupId>javax.xml.bind</groupId>
+		    <artifactId>jaxb-api</artifactId>
```
Comment (jdamerow) [inline vspace/pom.xml:197]: why are you adding jaxb?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 7 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:170]: ?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 8 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:253]: ?

**Review by jdamerow — CHANGES_REQUESTED** [review]
there are unaddressed comments

**Comment by jdamerow** [comment]
- image isn't full screen but bigger than the window size
- when the image doesn't change size (is too small), it jumps to the middle of the window when clicking the fullwidth button

File: vspace/pom.xml [hunk 12]
```diff
@@ -12,7 +12,7 @@
         <org.slf4j-version>1.7.5</org.slf4j-version>
         <spring.version>5.0.19.RELEASE</spring.version>
         <spring-data.version>Kay-SR8</spring-data.version>
-        <spring-security-version>5.3.10.RELEASE</spring-security-version>
+        <spring-security-version>5.3.3.RELEASE</spring-security-version>
```
Comment (jdamerow) [inline vspace/pom.xml:15]: is there a reason you downgraded this?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 7 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:173]: this variable should be declared inside the function

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 13]
```diff
@@ -134,6 +163,43 @@
...
+                    	$('#bgImage').width('100%');
+                    	$(".spaceClass").css("width", "80%");
+                    	$(".spaceClass").css("position", "relative");
+                    	$(".container-fluid").css("padding-right", "15px");
+                    	$(".container-fluid").css("padding-left", "15px");
+                	}
+                }
+		$("#space a").remove();        
+		drawLinks();
+            }
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:202]: formatting

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 8 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:263]: variable should be declared here

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/pom.xml (hunk 12 above)
Comment (pradeep-017) [reply vspace/pom.xml:15]: I think this has happened accidentally and committed. I have made changes now to correct this to latest.

**Comment by pradeep-017** [comment]
Resolved all comments!

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 14]
```diff
@@ -134,6 +163,41 @@
...
+                // resize the image to max width
+		if (isResizeImgBtnClicked == 1) {
+			$("#toggleImgClass").attr('class', "icon-collapse secondary");
+			if(maxImgWidthOrig > maxImgWidth) {
+				$("#bgImage").css("width", maxImgWidth);
+            	$(".spaceClass").css("width", "100%"); 
+            	$(".spaceClass").css("position", "");
+            	$(".container-fluid").css("padding-right", "0px");
+            	$(".container-fluid").css("padding-left", "0px");
+			}
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:185]: formatting

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 15]
```diff
@@ -134,6 +163,41 @@
...
+                }
+                else{
+                	$("#toggleImgClass").attr('class', "icon-expand secondary");
+                	if(maxImgWidthOrig > maxImgWidth) {
+                    	$("#bgImage").css("width", spaceWidth);
+                    	$('#bgImage').width('100%');
+                    	$(".spaceClass").css("width", "80%");
+                    	$(".spaceClass").css("position", "relative");
+                    	$(".container-fluid").css("padding-right", "15px");
+                    	$(".container-fluid").css("padding-left", "15px");
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:195]: formatting

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 16]
```diff
@@ -172,14 +236,39 @@
...
+		    }
+		    document.cookie = name + "=" + (value || "")  + expires + "; path=/";
+		}
+		
+		function getCookie(name) {
+		    var nameEQ = name + "=";
+		    var ca = document.cookie.split(';');
+		    for(var i=0;i < ca.length;i++) {
+		        var c = ca[i];
+		        while (c.charAt(0)==' ') c = c.substring(1,c.length);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:254]: is this for trimming whitespace? why not just use `trim()`?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 17]
```diff
@@ -172,14 +236,39 @@
...
+		        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:255]: why not use `startsWith()`? much easier to understand and read.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 18]
```diff
@@ -172,14 +236,39 @@
...
+		        var date = new Date();
+		        date.setTime(date.getTime() + (days*24*60*60*1000));
+		        expires = "; expires=" + date.toUTCString();
+		    }
+		    document.cookie = name + "=" + (value || "")  + expires + "; path=/";
+		}
+		
+		function getCookie(name) {
+		    var nameEQ = name + "=";
+		    var ca = document.cookie.split(';');
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:251]: bad variable name

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 19]
```diff
@@ -172,14 +236,39 @@
...
+		    for(var i=0;i < ca.length;i++) {
+		        var c = ca[i];
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:253]: bad variable name

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 20]
```diff
@@ -172,14 +236,39 @@
...
+		        while (c.charAt(0)==' ') c = c.substring(1,c.length);
+		        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
+		    }
+		    return null;
+		}
+		
 		$(window).on("load", function() {
+			var isResizeImgBtnClicked = getCookie('isResizeImgBtnClicked');
+			if(isResizeImgBtnClicked == "")
+				setCookie('isResizeImgBtnClicked', -1, null);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:263]: indentation and formatting (missing `{}`)

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 21]
```diff
@@ -172,14 +236,39 @@
...
 		    let spaceHeight = $("#bgImage").css("height");
 		    $("#Module_1").css("height",spaceHeight);
 			$( window ).resize(function() {
 				$("#space a").remove();
 		    	drawLinks();
 		    });
 		    drawLinks();
+		    toggleImgSize(0);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:271]: `toggleImgSize` expects a boolean, doesn't it. Pass true or false to make it easier to read/understand.

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- Image position has changed (it's too low now). Should be where it was:
image
- when the expand button is clicked, the menu moves to the left and the heading as well (only the image should be full screen)
- the resize button doesn't work once I'm on a new page (I clicked the full screen button then went to another page, the button doesn't work anymore)

**Comment by pradeep-017** [comment]
Fixed all review comments and now the functionality of making image full width works.

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html [hunk 22]
```diff
@@ -131,9 +161,54 @@
...
-    <div layout:fragment="content">
+    <div id="mainDivContent" layout:fragment="content">
         <script th:inline="javascript">
 		//# sourceURL=click.js
+			
+            function toggleImgSize(isClicked){
+                let maxImgWidth = $(window).width();
+                let spaceWidth = $("#bgImage").css("width");
+                let maxImgWidthOrig = parseFloat($("#bgImage").css("max-width"));
+                var isResizeImgBtnClicked = getCookie('isResizeImgBtnClicked');
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:172]: is there a reason you mix let and var here?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 14 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:193]: ?

File: vspace/src/main/webapp/WEB-INF/views/exhibition/space.html (hunk 13 above)
Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/exhibition/space.html:211]: ?

**Review by jdamerow — CHANGES_REQUESTED** [review]
there are several unaddressed comments; closing until they are addressed.

**Review by jdamerow — CHANGES_REQUESTED** [review]
There are unaddressed comments. Please don't reopen until all comments have been addressed.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**